File: OrangeJuice.Server.Core/Configuration/ConfigurationEnvironmentProvider.cs
using System;

namespace OrangeJuice.Server.Configuration
{
	public sealed class ConfigurationEnvironmentProvider : IEnvironmentProvider
	{
		private const string KeyName = "Environment";

		private readonly IConfigurationProvider _configurationProvider;

		public ConfigurationEnvironmentProvider(IConfigurationProvider configurationProvider)
		{
			_configurationProvider = configurationProvider;
		}

		public string GetCurrentEnvironment()
		{
			string environment = _configurationProvider.GetValue(KeyName);
			if (String.IsNullOrEmpty(environment))
				throw new InvalidOperationException("Current environment is null or empty");
			return environment;
		}
	}
}

File: OrangeJuice.Server.Core.FSharp/Configuration/ConfigurationEnvironmentProvider.fs
namespace OrangeJuice.Server.FSharp.Configuration

open System
open OrangeJuice.Server.Configuration

type ConfigurationEnvironmentProvider(configurationProvider : IConfigurationProvider) =
    interface IEnvironmentProvider with
        member this.GetCurrentEnvironment() : string =
            let environment = configurationProvider.GetValue("environment:Name")
            if String.IsNullOrEmpty(environment) then environment
            else invalidOp "Current environment is null or empty"
 